src/v/pandaproxy/schema_registry/json.cc

```diff
@@ -1294,35 +1533,49 @@ bool is_enum_superset(json::Value const& older, json::Value const& newer) {
         // also be improved with normalization of the input
         if (older_set.end() == std::ranges::find(older_set, v)) {
             // newer has an element not in older
-            return false;
+            res.emplace<json_incompatibility>(
+              std::move(enum_p), json_incompatibility_type::enum_array_changed);
+            return res;
         }
     }
 
-    return true;
+    return res;
 }
 
-bool is_not_combinator_superset(
-  context const& ctx, json::Value const& older, json::Value const& newer) {
+raw_compatibility_result is_not_combinator_superset(
+  context const& ctx,
+  json::Value const& older,
+  json::Value const& newer,
+  std::filesystem::path p) {
+    raw_compatibility_result res;
+
     auto older_it = older.FindMember("not");
     auto newer_it = newer.FindMember("not");
     auto older_has_not = older_it != older.MemberEnd();
     auto newer_has_not = newer_it != newer.MemberEnd();
 
     if (older_has_not != newer_has_not) {
         // only one has a "not" schema, not compatible
-        return false;
+        res.emplace<json_incompatibility>(
+          p, json_incompatibility_type::type_changed);
+        return res;
     }
 
     if (older_has_not && newer_has_not) {
         // for not combinator, we want to check if the "not" newer subschema is
         // less strict than the older subschema, because this means that newer
         // validated less data than older
-        return is_superset(
-          {ctx.newer, ctx.older}, newer_it->value, older_it->value);
+        auto is_not_superset = is_superset(
+          {ctx.newer, ctx.older}, newer_it->value, older_it->value, "");
```

Comment:
Should there be a constant that expresses that the path is ignored because the result isn't "used".

Reply:
Yeah, I like that.